src/AbstractClient.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JTL\GoPrometrics\Client;
+
+use GuzzleHttp\ClientInterface;
+
+abstract class AbstractClient
+{
+    private ?ClientInterface $client;
+    protected string $baseUrl = '';
+
+    public function __construct(
+        ?ClientInterface $client = null,
+        string $baseUrl = ''
+    ) {
+        $this->client = $client;
+        $this->baseUrl = $baseUrl;
+    }
+
+    public function getClient(): ?ClientInterface
```

Comment:
Why exposing the ClientInterface in here or do you see any use case to send this class around.

Would suggest to keep it private

Reply:
Yeah, you right and it's gone.